File: Remotion/Data/Linq.IntegrationTests.CSharp/LinqSamples101/GroupTopBottom.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Remotion.Data.Linq.IntegrationTests.Utilities;

namespace Remotion.Data.Linq.IntegrationTests.CSharp.LinqSamples101
{
  class GroupTopBottom:Executor
  {
    //This sample uses Take to select the first 5 Employees hired.")]
    public void LinqToSqlTop01 ()
    {
      var q = (
          from e in db.Employees
          orderby e.HireDate
          select e)
          .Take (5);

      serializer.Serialize (q);
    }

    //This sample uses Skip to select all but the 10 most expensive Products.")]
    public void LinqToSqlTop02 ()
    {
      var q = (
          from p in db.Products
          orderby p.UnitPrice descending
          select p)
          .Skip (10);

      serializer.Serialize (q);
    }
  }
}

File: Remotion/Data/Linq.IntegrationTest.VisualBasic/LinqSamples101/GroupTopBottom.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.
Option Infer On
Option Strict On

Imports System.Collections.Generic
Imports System.Data
Imports System.Data.SqlClient
Imports System.IO
Imports System.Linq
Imports System.Linq.Expressions
Imports System.Reflection
Imports Remotion.Data.Linq.IntegrationTests.Utilities

Namespace LinqSamples101
  Public Class GroupTopBottom
    Inherits Executor
    '<Category("TOP/BOTTOM")> _
    '  <Title("Take")> _
    '  <Description("This sample uses Take to select the first 5 Employees hired.")> _
    Public Sub LinqToSqlTop01()
      Dim first5Employees = From emp In db.Employees _
                            Order By emp.HireDate _
                            Take 5

      serializer.Serialize(first5Employees)
    End Sub

    '<Category("TOP/BOTTOM")> _
    '<Title("Skip")> _
    '<Description("This sample uses Skip to select all but the 10 most expensive Products.")> _
    Public Sub LinqToSqlTop02()
      Dim expensiveProducts = From prod In db.Products _
                              Order By prod.UnitPrice Descending _
                              Skip 10

      serializer.Serialize(expensiveProducts)
    End Sub
  End Class
End Namespace

